exchanges/stream/websocket_test.go

```diff
@@ -1050,22 +1201,20 @@ func TestFlushChannels(t *testing.T) {
 	err = w.FlushChannels()
 	require.NoError(t, err, "Flush Channels must not error")
 
-	w.features.FullPayloadSubscribe = true
-	w.GenerateSubs = problemFunc
-	err = w.FlushChannels() // error on full subscribeToChannels
+	w.GenerateSubs = func() (subscription.List, error) { return nil, errDastardlyReason } // error on generateSubs
+	err = w.FlushChannels()                                                               // error on full subscribeToChannels
 	assert.ErrorIs(t, err, errDastardlyReason, "FlushChannels should error correctly on GenerateSubs")
 
-	w.GenerateSubs = noSub
-	err = w.FlushChannels() // No subs to sub
+	w.GenerateSubs = func() (subscription.List, error) { return nil, nil } // No subs to sub
+	err = w.FlushChannels()                                                // No subs to sub
```

Comment:
```
    websocket_test.go:1210: 
        	Error Trace:	/Users/scottg/go/src/github.com/thrasher-corp/gocryptotrader/exchanges/stream/websocket_test.go:1210
        	Error:      	Received unexpected error:
        	            	test websocket already connected
        	Test:       	TestFlushChannels
        	Messages:   	Flush Channels should not error
```

I got this, then also debugged it. So it kinda goes through:
- FlushChannels() 
    - Connect()
       -  Hits"
   ```go
	if w.IsConnected() {
		return fmt.Errorf("%v %w", w.exchangeName, errAlreadyConnected)
	}
    ```

My guess is the connection monitor kicks in too quickly. Please just sanity check to see that its not going to cause problems for things. For the test itself is likely to increase the timer

Reply:
It also comes up in Actions: 
https://github.com/thrasher-corp/gocryptotrader/actions/runs/10950367635/job/30405470763?pr=1580#step:8:313
https://github.com/thrasher-corp/gocryptotrader/actions/runs/10950367635/job/30405471659?pr=1580#step:10:389

Reply:
Another Actions test death:
https://github.com/thrasher-corp/gocryptotrader/actions/runs/10950367635/job/30405471508?pr=1580#step:10:395